src/control/server/ownership.go

```diff
@@ -135,3 +131,32 @@ func changeFileOwnership(config *configuration) error {
 
 	return nil
 }
+
+// respawnProc builds and executes command chain to respawn process under new ownership.
+func respawnProc(config *configuration) error {
```

Comment:
Why are these two very different operations stuck together?

From a clarity, testing, and maintenance perspective, it seems better to have a function which will do the NVMe prep, then a separate function to fork/exec with lowered privileges.

Reply:
we need to do both after this process dies, therefore they need to be change together and executed independently after this process has been able to exit

Reply:
Please elaborate on this. Why does storage prep have to happen after the parent daos_server process dies?

Reply:
because the handles that the running process has open on SPDK libraries prevent storage-prep from doing what needs to and those references can't be completely cleaned up during the process lifetime.

Reply:
more details in https://jira.hpdd.intel.com/browse/DAOS-2644